### ISSUE TITLE
Checking for AttributeError in tests

### ISSUE THREAD

**Issue by rpoleski** [rpoleski/MulensModel#149]
First, I must admit that @jenniferyee nailed it! In Version3_1 file tests/test_ModelParameters.py has:

```python
    def test_basic_1L3S_rho(self):
...
        # JCY: don't understand why this test doesn't work, because it does throw an Attribute Error.
        # with np.testing.assert_raises(AttributeError):
        #    print(self.model_params.rho_1)
```

I've made a MWE:

```python
import unittest
import MulensModel as mm

class Test_1(unittest.TestCase):
    def test_A(self):
        model_params = mm.ModelParameters({
            't_0_1': 1., 'u_0_1': 0.5,
            't_0_2': 1., 'u_0_2': 0.5, 'rho_2': .1,
            't_E': 3})
        with self.assertRaises(AttributeError):
            print(model_params.rho_1)

    def test_B(self):
        model_params = mm.ModelParameters({
            't_0_1': 1., 'u_0_1': 0.5,
            't_0_2': 1., 'u_0_2': 0.5, 'rho_2': .1,
            't_0_3': 2., 'u_0_3': 0.3, 't_star_3': 0.02,
            't_E': 3})
        with self.assertRaises(AttributeError):
            print(model_params.rho_1)
```
The 2 tests differ by only one line (`'t_0_3': ...`) yet the first one passes and the second fails. Any idea what's wrong?

**Comment by rpoleski** [comment]
Ok, it took me a moment to find likely culprit that makes difference between 2 and 3 sources. @rapoliveira comment to @jenniferyee commit was also a clue. The culprit is:

```python
    def __getattr__(self, item):
        if self.n_sources <= 2:
            return object.__getattribute__(self, item)
        else:
            for source_param in ModelParameters._all_source_params_head:
                if item[0:len(source_param)] == source_param:
                    source_num = item.split('_')[-1]
                    try:
                        return self.__getattr__(
                            '_source_{0}_parameters'.format(
                                source_num)).__getattribute__(source_param)
                    except AttributeError:
                        return object.__getattribute__(self, item)

            if (len(item) > 6) and item[0:7] == 'source_':
                return object.__getattribute__(self, '_{0}'.format(item))

            return object.__getattribute__(self, item)
```

**Comment by rpoleski** [comment]
I've spend some time understanding what's happening. The solution is to change the last line of:
```python
    @property
    def rho(self):
        if 'rho' in self.parameters.keys():
        ...
        else:
            return None
```
to
```python
            raise AttributeError("rho is not defined and cannot be calculated")
```

Before finding this, I've written different getattr():
```python
    def __getattr__(self, item):
        end = item.split('_')[-1]
        if end.isnumeric() and int(end) > 0:
            head = item[:-len(end)-1]
            return self.__getattr__('_source_{:}_parameters'.format(end)).__getattribute__(head)
        elif item.startswith("source_") and item.endswith("_parameters"):
            return object.__getattribute__(self, "_" + item)
        else:
            return object.__getattribute__(self, item)
```
At the end I've found that it works the same as current version, but I think it's simpler. @jenniferyee do you agree to replace getattr()?

**Comment by jenniferyee** [comment]
@jenniferyee do you agree to replace getattr()?
Yes.

And also the change to rho.

**Comment by rpoleski** [comment]
Solved in 484adca8
